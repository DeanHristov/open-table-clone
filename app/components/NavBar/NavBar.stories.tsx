import * as React from "react";
import { Meta, Story } from "@storybook/react";
import NavBar, { INavBarProps } from "./NavBar";

export default {
  title: "Components/NavBar",
  component: NavBar,
} as Meta;

export const Default: Story<INavBarProps> = (args) => <NavBar {...args} />;

Default.args = {};
