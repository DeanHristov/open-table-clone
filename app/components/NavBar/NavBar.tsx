"use client";

import { FC } from "react";
import Logo from "../Logo";
import Button from "../Button";
import { MagnifyingGlassIcon } from "@heroicons/react/24/solid";

export interface INavBarProps {}

const NavBar: FC<INavBarProps> = ({}) => {
  return (
    <nav className="bg-white p-1 flex justify-between">
      <Logo />
      <div className="flex">
        <Button blue title="Sign in" />
        <Button title="Sign up" />
        <div className="flex items-center px-2">
          <MagnifyingGlassIcon className="h-6 w-6 text-black inline-block cursor-pointer" />
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
