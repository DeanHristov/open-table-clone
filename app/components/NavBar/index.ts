import NavBar, { INavBarProps } from "./NavBar";

export type { INavBarProps };
export default NavBar;
