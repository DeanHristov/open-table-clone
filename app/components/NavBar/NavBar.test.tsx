import { render } from "@testing-library/react";
import NavBar from "./NavBar";

describe("Component: <NavBar {...} />", () => {
  it("Should be able to render default state", () => {
    const { container } = render(<NavBar />);

    expect(container).toMatchSnapshot();
  });
});
