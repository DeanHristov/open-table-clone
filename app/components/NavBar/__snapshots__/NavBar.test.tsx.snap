// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <NavBar {...} /> Should be able to render default state 1`] = `
<div>
  <nav
    class="bg-white p-1 flex justify-between"
  >
    <div
      class="m-0 p-0"
    >
      <a
        href="/"
      >
        <img
          alt="Open table"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/opentable-logo.svg"
          style="color: transparent;"
          width="200"
        />
      </a>
    </div>
    <div
      class="flex"
    >
      <div
        class="p-1 h-auto"
      >
        <button
          class="rounded w-full px-4 py-2 text-center text-sm transition-all duration-300 hover:shadow-md cursor-pointer disabled:cursor-not-allowed bg-shades-100 text-white hover:bg-shades-200"
        >
           
          Sign in
        </button>
      </div>
      <div
        class="p-1 h-auto"
      >
        <button
          class="rounded w-full px-4 py-2 text-center text-sm transition-all duration-300 hover:shadow-md cursor-pointer disabled:cursor-not-allowed bg-white hover:bg-gray-50 border text-black"
        >
           
          Sign up
        </button>
      </div>
      <div
        class="flex items-center px-2"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6 text-black inline-block cursor-pointer"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.5 3.75a6.75 6.75 0 100 13.5 6.75 6.75 0 000-13.5zM2.25 10.5a8.25 8.25 0 1114.59 5.28l4.69 4.69a.75.75 0 11-1.06 1.06l-4.69-4.69A8.25 8.25 0 012.25 10.5z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </nav>
</div>
`;
